scale-encode/src/impls/mod.rs

```diff
@@ -1215,4 +1235,65 @@ mod test {
             123u64,
         );
     }
+
+    #[test]
+    fn smart_pointer_fields() {
```

Comment:
Nit: clarify the name a touch? :)

```suggestion
    fn encode_smart_pointers_as_fields() {
```